src/main/scala/org/scalacheck/MasterRunner.scala

```diff
@@ -0,0 +1,67 @@
+package org.scalacheck
+
+
+import sbt.testing._
+import java.util.concurrent.atomic.AtomicInteger
+
+class  MasterRunner(
+                    args: Array[String],
+                    remoteArgs: Array[String],
+                    loader: ClassLoader)
+  extends BaseRunner(args, remoteArgs, loader) {
+
+
+  /** Number of tasks registered in the whole system */
+  private[this] val registeredCount = new AtomicInteger(0)
+
+  /** Number of running slaves in the whole system */
+  private[this] val slaveCount = new AtomicInteger(0)
+
+  def tasks(taskDefs: Array[TaskDef]): Array[Task] = {
+    registeredCount.addAndGet(taskDefs.length)
+    taskDefs.map(newTask)
+  }
+
+  def done(): String = {
+    val slaves = slaveCount.get
+    val registered = registeredCount.get
+    val done = results.doneCount.get
+    val test = results.testCount.get
+    val success = results.successCount.get
+    val failed = results.failureCount.get
+    val errors = results.errorCount.get
+
+    if (slaves > 0) {
+      throw new IllegalStateException(
+        s"There are still $slaves slaves running")
+    }
```

Comment:
I gather you have this code (and the `registered != done` below) because you mimicked the testing framework made for testing testing (double intended) frameworks in Scala.js, aka https://github.com/scala-js/scala-js/blob/master/sbt-plugin-test/testFramework/src/main/scala/sbttest/framework/MasterRunner.scala

If yes, then you should simply remove all of this. We have this in our dummy testing frameworks to test our implementation of the sbt test interface, but in real testing frameworks, this is useless.

Which means you can get rid of `slaveCount`, `registeredCount`, `doneCount`, and the messages that maintain these.